### PR TITLE
Feature: ERC 20 connector

### PR DESCRIPTION
**Pull request by isra67**

### CODE REVIEW

File: src/model/request/BurnEthErc20.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    IsNotEmpty,
+    IsNumberString,
+    IsOptional,
+    Length,
+    Min,
+} from 'class-validator';
+import { Currency } from './Currency';
+import { BurnErc20 } from './BurnErc20'
+
+export class BurnEthErc20 extends BurnErc20 {
```
Comment (tatumio) [inline src/model/request/BurnEthErc20.ts:13]: this class is not needed. we can use Base class for ETH

File: src/model/request/DeployEthErc20.ts [hunk 2]
```diff
@@ -11,39 +11,7 @@ import {
...
-
-    @IsOptional()
-    @Min(0)
-    public nonce?: number;
-
-    @IsOptional()
-    @ValidateNested()
-    @Type(() => Fee)
-    public fee?: Fee;
+export class DeployEthErc20 extends DeployErc20 {
```
Comment (tatumio) [inline src/model/request/DeployEthErc20.ts:16]: this class is not needed. we can use Base class for ETH

File: src/transaction/celo.ts [hunk 3]
```diff
@@ -22,6 +22,17 @@ import {
     TransferCeloOrCeloErc20Token,
 } from '../model';
 
+/**
+ * Returns server to connect to.
+ *
+ * @param provider url of the Server to connect to. If not set, default public server will be used.
+ * @param fromPrivateKey optional private key of sender account
+ */
+ export const getCeloClient = (provider?: string, fromPrivateKey?: string): CeloProvider => {
```
Comment (tatumio) [inline src/transaction/celo.ts:31]: why is it here?

File: src/transaction/eth.ts [hunk 4]
```diff
@@ -116,59 +118,144 @@ export const prepareStoreDataTransaction = async (body: CreateRecord, provider?:
...
+    fromPrivateKey,
+    amount,
+    to,
+    contractAddress,
+    nonce,
+    signatureId,
+  } = body;
+
+  const client = getClient(provider);
+  client.eth.accounts.wallet.clear();
```
Comment (tatumio) [inline src/transaction/eth.ts:138]: these 3 lines should not be here - they are in the getClient method

File: src/transaction/eth.ts [hunk 5]
```diff
@@ -116,59 +118,144 @@ export const prepareStoreDataTransaction = async (body: CreateRecord, provider?:
...
+  const {
+    fromPrivateKey,
+    amount,
+    contractAddress,
+    nonce,
+    signatureId,
+  } = body;
+
+  const client = getClient(provider);
+  client.eth.accounts.wallet.clear();
```
Comment (tatumio) [inline src/transaction/eth.ts:180]: these 3 lines should not be here - they are in the getClient method

**Review by tatumio — CHANGES_REQUESTED** [review]

File: src/transaction/eth.ts (hunk 4 above)
Comment (isra67) [reply src/transaction/eth.ts:138]: this sequence is used everywhere in the file

File: src/model/request/BurnEthErc20.ts (hunk 1 above)
Comment (isra67) [reply src/model/request/BurnEthErc20.ts:13]: ok

File: src/model/request/DeployEthErc20.ts (hunk 2 above)
Comment (isra67) [reply src/model/request/DeployEthErc20.ts:16]: ok

File: src/transaction/celo.ts (hunk 3 above)
Comment (isra67) [reply src/transaction/celo.ts:31]: ok

**Review by tatumio — APPROVED** [review]

**Pull request merged**